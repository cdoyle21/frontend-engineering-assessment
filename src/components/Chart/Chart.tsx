import React, { Component} from 'react';
import Aux from '../../hoc/Aux';
import Buttons from '../Buttons/Buttons';
import ParentSize from '@visx/responsive/lib/components/ParentSize';
import TopicByMonthChart from '../Chart/TopicByMonthChart/TopicByMonthChart';
import UserProfileTopicsChart from '../Chart/UserProfileTopicsChart/UserProfileTopicsChart';

class TopicByMonth extends Component {
    render () {
        return (
            <Aux>
                <ParentSize>{({ width, height }) => <TopicByMonthChart width={width} height={height} />}</ParentSize>
                <ParentSize>{({ width, height }) => <UserProfileTopicsChart width={width} height={height} />}</ParentSize>
                <Buttons>TOPICS BY MONTH</Buttons>
            </Aux>
        )
    }
}

export default TopicByMonth;