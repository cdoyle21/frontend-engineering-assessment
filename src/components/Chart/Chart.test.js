import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import React from 'react';
import Chart from './Chart';

configure({ adapter: new Adapter() });

describe('Render Chart component', () => {
    it('should be defined', () => {
        expect(Chart).toBeDefined();
      });
    it('should render Chart component correctly', () => {
        const wrapper = shallow(
        <Chart />,
        );
        expect(wrapper).toMatchSnapshot();
    });
  });
