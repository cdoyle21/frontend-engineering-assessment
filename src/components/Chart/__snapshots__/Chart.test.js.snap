// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Chart component should render Chart component correctly 1`] = `ShallowWrapper {}`;
